kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveKafkaClientSupplier.java

```diff
@@ -130,7 +131,7 @@ public Consumer<byte[], byte[]> getConsumer(final Map<String, Object> config) {
     );
     // TODO: the end offsets poller call is kind of heavy for a synchronized block
     return factories.createResponsiveConsumer(
```

Comment:
@rodesai sorry if this is a dumb question or if I asked it before, because I' remember being confused by this before, but I really don't understand the point of the `factories` approach in this class -- AFAICT there's just the one, default/empty implementation of the interface and everything just passes the input parameters directly into the constructors. Is this intended for some future feature?

Followup question: assuming there is a reason for this, why don't we have a factory for all the clients? eg why the main consumer but not the restore?

Reply:
It's for letting me inject mocks when unit testing to make sure that the right listeners get installed for the right threads' clients, and to test that the ref counting works.